ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpService.java

```diff
@@ -809,4 +820,51 @@ private String buildCorsRegexFromConfig() {
       return stringJoiner.toString();
     }
   }
+
+  static class JsonResponseStreamer extends OutputStream {
+    private final HttpServerResponse response;
+    private final Semaphore paused = new Semaphore(0);
+    private final byte[] singleByteBuf = new byte[1];
+    private boolean chunked = false;
+
+    public JsonResponseStreamer(final HttpServerResponse response) {
+      this.response = response;
+    }
+
+    @Override
+    public void write(final int b) throws IOException {
+      singleByteBuf[0] = (byte) b;
+      write(singleByteBuf, 0, 1);
+    }
+
+    @Override
+    public void write(final byte[] bbuf, final int off, final int len) throws IOException {
+      if (!chunked) {
+        response.setChunked(true);
+        chunked = true;
+      }
+
+      if (response.writeQueueFull()) {
+        LOG.debug("HttpResponse write queue is full pausing streaming");
+        response.drainHandler(e -> paused.release());
+        try {
+          paused.acquire();
+          LOG.debug("HttpResponse write queue is not accepting more data, resuming streaming");
```

Comment:
Is this meant to be '... is now accepting more data' ?

Reply:
yes, this is another benefit of streaming the response with Vert.x, since it is possible to implement a back pressure strategy, to avoid sending more data than the client can receive, and so less memory will be used because the JSON fragments are created only when they can be sent to the client, instead of creating all the JSON string upfront.